### PR TITLE
deprecated `react-utilties`

### PR DESCRIPTION
**Pull request by michelleyschen**
### WHY are these changes introduced?

`react-utilties` is now [deprecated](https://github.com/Shopify/react-utilities/pull/34)
This PR remove the use of this library by doing the following
 
-  Replace `className` & `variationName` from `@shopify/react-utilities/styles` with `@shopify/css-utilities`
- Replace `ReactComponent` from `@shopify/react-utilities` with `React.ComponentType`
- Move `wrapWithComponent`, `isElementOfType` and `elementChildren` &  from `@shopify/react-utilities/styles/components` to `src/utilities/components.tsx`
- Move `isServer`  from `@shopify/react-utilities/styles/target` to `src/utilities/target.ts`

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {Page} from '@shopify/polaris';

interface State {}

export default class Playground extends React.Component<never, State> {
  render() {
    return (
      <Page title="Playground">
        {/* Add the code you want to test here */}
      </Page>
    );
  }
}
```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: src/components/AppProvider/utilities/createAppProviderContext/tests/createAppProviderContext.test.tsx [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import * as React from 'react';
 import * as appBridge from '@shopify/app-bridge';
 import {noop} from '@shopify/javascript-utilities/other';
-import * as targets from '@shopify/react-utilities/target';
+
+import * as targets from '../../../../../utilities/target';
```
Comment (BPScott) [inline src/components/AppProvider/utilities/createAppProviderContext/tests/createAppProviderContext.test.tsx]: `import * as ....` is [on its way out to enable compiling TS with babel directly](https://github.com/Shopify/web/pull/13571).

I know this isn't strictly in the scope of this PR but would you be able to look at working out how we can avoid doing  an `import * as targets` for the tests that mock the result of isServer?

File: src/components/Avatar/tests/Avatar.test.tsx [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import * as React from 'react';
-import * as targets from '@shopify/react-utilities/target';
 import {mountWithAppProvider, trigger} from 'test-utilities';
 import {Image} from 'components';
+
+import * as targets from '../../../utilities/target';
```
Comment (BPScott) [inline src/components/Avatar/tests/Avatar.test.tsx]: Can we dodge this `import *`?

File: src/components/Portal/tests/Portal.test.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import * as React from 'react';
-import * as targets from '@shopify/react-utilities/target';
 import {mountWithAppProvider} from 'test-utilities';
+import * as targets from '../../../utilities/target';
```
Comment (BPScott) [inline src/components/Portal/tests/Portal.test.tsx]: Can we dodge this `import *`?

File: src/components/Portal/tests/Portal.test.tsx [hunk 4]
```diff
@@ -12,38 +11,21 @@ const {
...
-  });
-
   describe('children', () => {
     it('get passed into the portal creation method', () => {
       const children = <div />;
       mountWithAppProvider(<Portal>{children}</Portal>);
       expect(createPortalSpy).toHaveBeenCalledWith(children, expect.anything());
     });
-
-    it('dont get rendered when the env is the server', () => {
```
Comment (michelleyschen) [inline src/components/Portal/tests/Portal.test.tsx]: the `isServer` check is no longer part of the `Portal` component?

**Comment by danrosenthal** [comment]
Looking at our consumer build tests, and it appears this might be responsible for failures there: https://buildkite.com/shopify/polaris-react/builds/729#cb6d7e9a-b2ef-40a9-864a-73867eb586ed

cc: @bpscott

**Comment by BPScott** [comment]
Looks like all of that is down to a component in web trying to import from react-utilities. Web doesn’t have an explicit dependency on react-utilities, and now that it’s no longer available transitively it’s exploding. This is why projects should explicitly state their deps rather than rely on transitive dependencies.

Solution is to add an explicit dependency on react-utilities in web
Longer term solution is to fixup web so it doesn't depend on react-utilities

**Comment by michelleyschen** [comment]
This is resolved now. 
https://buildkite.com/shopify/polaris-react/builds/733

**Pull request merged**